cmd/list/machinepool/cmd.go

```diff
@@ -28,38 +29,49 @@ import (
 	"github.com/openshift/rosa/pkg/rosa"
 )
 
-var Cmd = &cobra.Command{
-	Use:     "machinepools",
-	Aliases: []string{"machinepool", "machine-pools", "machine-pool"},
-	Short:   "List cluster machine pools",
-	Long:    "List machine pools configured on a cluster.",
-	Example: `  # List all machine pools on a cluster named "mycluster"
-  rosa list machinepools --cluster=mycluster`,
-	Run:  run,
-	Args: cobra.NoArgs,
-}
+const (
+	use     = "machinepools"
+	short   = "List cluster machine pools"
+	long    = "List machine pools configured on a cluster."
+	example = `  # List all machine pools on a cluster named "mycluster"
+  rosa list machinepools --cluster=mycluster`
+)
 
-func init() {
-	ocm.AddClusterFlag(Cmd)
-	output.AddFlag(Cmd)
-}
+var (
+	aliases = []string{"machinepool", "machine-pools", "machine-pool"}
+)
 
-func run(_ *cobra.Command, _ []string) {
-	r := rosa.NewRuntime().WithAWS().WithOCM()
-	defer r.Cleanup()
+func NewListMachinePoolCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     use,
+		Short:   short,
+		Long:    long,
+		Aliases: aliases,
+		Example: example,
+		Args:    cobra.MaximumNArgs(1),
+		Run:     rosa.DefaultRunner(rosa.RuntimeWithOCM(), ListMachinePoolRunner()),
+	}
+
+	output.AddFlag(cmd)
+	ocm.AddClusterFlag(cmd)
+	return cmd
+}
 
-	clusterKey := r.GetClusterKey()
+func ListMachinePoolRunner() rosa.CommandRunner {
+	return func(_ context.Context, runtime *rosa.Runtime, cmd *cobra.Command, _ []string) error {
+		clusterKey := runtime.GetClusterKey()
 
-	cluster := r.FetchCluster()
-	if cluster.State() != cmv1.ClusterStateReady &&
-		cluster.State() != cmv1.ClusterStateHibernating {
-		r.Reporter.Errorf("Cluster '%s' is not yet ready", clusterKey)
-		os.Exit(1)
-	}
+		cluster := runtime.FetchCluster()
+		if cluster.State() != cmv1.ClusterStateReady &&
+			cluster.State() != cmv1.ClusterStateHibernating {
+			return fmt.Errorf("cluster '%s' is not yet ready", clusterKey)
```

Comment:
I know there is warning in the editor that error message should not start with Capital letter. But imo that's not a good UX for end user considering our usage.
I am ok either way if all of us follow a consistent code style. Just want to make sure which one we select.

Reply:
hmm, i did lower case on the last refactor, but looking at ROSA, i'd say majority is upper case. Will change this, thanks marco

Reply:
Changed these errors + the `describe machinepool` errors from the last refactor